app/controllers/api/mentor_discussion_posts_controller.rb

```diff
@@ -35,6 +35,18 @@ def create
       head 200
     end
 
+    def update
+      post = Solution::MentorDiscussionPost.find_by(uuid: params[:id], author: current_user)
```

Comment:
Could you change this so that it just finds by uuid, then we check the author pls? The reason is that if we just look by the uuid, we can use the index which will be faster.